picard/util/__init__.py

```diff
@@ -486,7 +495,7 @@ def load_json(data):
         dict: Response data as a python dict
 
     """
-    return json.loads(convert_to_string(data))
+    return json.loads(bytes(data).decode())
```

Comment:
Can we really be sure to always get byte arrays here? I have checked the plugins, the current plugins that use it also just use it for webservice results. But this function actually could be handy for also loading JSON from string. I would keep it flexible to accept both.

Reply:
This is one of the reasons I introduced string_

Reply:
There will also be problems with the web browser module.

Reply:
Ok, i'll fix it, especially since the description specifies it ... ;)

Reply:
> There will also be problems with the web browser module.

Which problems exactly?

Reply:
It should mostly get called with `QByteArray`, because it is always used with the results from the webservice module. But I would still make it also accept strings in this case because of unknown plugin usage.

Reply:
The web browser module is used in different contexts iirc. Some of the times it has to handle bytes, sometimes qbytesarray and sometimes string. Check it's usage instances, idr all of it. But it was one of the primary reasons I introduced this dynamic conversion.

Reply:
> But I would still make it also accept strings in this case because of unknown plugin usage.

Whatever we change, that's likely to break an "unknown plugin" ... and i think it's a bad thing.
We definitively need to find a good solution for this.

For now, i'd rather bump plugin API version, rather that not doing change in core code...

Reply:
I frankly think this is premature optimization to break compatibility. Did we actually measure the performance degradation due to this call in normal picard usage and if it was significant? If it is not, I'd rather keep it in the api and webbrowser module. Also just by keeping the built-in I don't think we are hurting anybody, it's just 4 loc. We can add a warning to not use it in the future.

Reply:
I don't think it is primarily a performance issue, but rather the frequent use of `string_` hides what data you are dealing with. And because at some point you stop knowing what you are handling you start adding string_ to even more code.

But for this specific case of `load_json` I agree that we should keep the string compatibility.

Reply:
Apart "unknown plugin" case, i don't think `json_load()` is used with anything but `bytes`.

So i'd rather move the "compatibility" to the call in `picard/plugin.py`.

And no, that's not a performance issue, just a bad idea that needs to be addressed before it's too late.

Reply:
You don't have to add anything to the call in `picard/plugin.py`, that one is fine. I'm also ok with leaving this as is, since all calls to it (including inside plugins) I could find actually pass bytes.